src/main/java/htsjdk/samtools/util/StringUtil.java

```diff
@@ -545,4 +545,56 @@ public static int levenshteinDistance(final String string1, final String string2
 
         return i;
     }
+
+    /**
+     * Calculates the hamming distance between two strings s1 and s2.
+     * Since Hamming distance is not defined for strings of differing lengths, we throw an exception if
+     * the two strings are of different lengths.
+     *
+     * @param s1 The first string to compare
+     * @param s2 The second string to compare, note that if s1 and s2 are swapped the value returned will be identical.
+     * @return Hamming distance between s1 and s2.
+     * @throws IllegalArgumentException If the two strings have differing lengths.
+     */
+    public static int hammingDistance(final String s1, final String s2) {
```

Comment:
Since likely this is going to be used to compare DNA sequences, you probably want in the doc stating that the comparison is case sensitive.  Also, do you have a need to count or not count `N`s as mismatches?